### ISSUE TITLE
💡Github pages does not render README file correctly

### ISSUE THREAD

**Issue by Incogdino** [nus-cs2103-AY2425S1/forum#302]
As I was trying to set up GIthub Pages for my repo, I came across an issue where the README does not render properly on Github Pages. For example, this was what was produced when setting it up

![image](https://github.com/user-attachments/assets/b086d4f5-a7f2-45d0-8f24-de4972e39627)

According to [Github Pages](https://docs.github.com/en/pages/setting-up-a-github-pages-site-with-jekyll/setting-a-markdown-processor-for-your-github-pages-site-using-jekyll), it supports 2 markdown processors, kramdown and Github Flavoured Markdown (GFM) 

Sadly for some reason that I do not know why, Github Pages uses kramdown. This can cause some formatting issue when the page is rendered. To solve this, you can follow this guide [here](https://docs.github.com/en/pages/setting-up-a-github-pages-site-with-jekyll/setting-a-markdown-processor-for-your-github-pages-site-using-jekyll).

Basically, you need to set up a `_config.yml` file under the `/docs` directory manually and specify the markdown processor
```yml
title: <title>
description: <description>
theme: <theme>
markdown: GFM
```

Even after setting this there are still some format that cannot be rendered properly. For example, 
- Task lists don't work as expected 
![image](https://github.com/user-attachments/assets/35fd8187-2d28-4c5f-b4d7-868d018afc1c)
- Alerts also do not render properly (Image taken from [here](https://github.com/github/docs/issues/27919))
![image](https://github.com/user-attachments/assets/2d67ad07-1a92-4262-a99b-57923112ff19)

If anyone has any suggestions, please share them here. I'm also unsure if this should be labelled as a tip 😃

**Comment by Dinoman44** [comment]
Is it directly using the `docs/README.md` file from your repository? If it isn't, you may have typo-ed something. From the screenshot you shared, it seems like the table is entirely missing and there is just a closing `</table>` tag. That might be causing parsing issues and may be the reason your markdown is not being rendered.

<img width="806" alt="Screenshot 2024-09-17 at 08 09 10" src="https://github.com/user-attachments/assets/a7b245e8-71cc-4344-bdc2-b9b82c9c1d65">

If that still doesn't work, try to use pipe characters for your table instead of using the `<table>` tags. [Here is how this works](https://diplodoc.com/docs/en/syntax/tables/gfm).

Basically just try to look for some inconsistency - unpaired brackets/backticks/tags, extra space etc.

**Comment by damithc** [comment]
@Incogdino Another thing: it will not be rendered properly if you don't separate markdown elements with a blank line.

Example A:
```
...
</table>
## Features
...
```

Example B:
```
...
</table>

## Features
...
```

Example A will not work but B will work.
The blank line is required by proper Markdown but the lack of it is forgiven by GitHub Markdown preview. So Example A will work in the GitHub Markdown preview (giving the wrong impression that all is well) but will not work when it is rendered into an HTML page via GitHub pages.

**Comment by Incogdino** [comment]
> @Incogdino Another thing: it will not be rendered properly if you don't separate markdown elements with a blank line.
> 
> Example A:
> 
> ```
> ...
> </table>
> ## Features
> ...
> ```
> 
> Example B:
> 
> ```
> ...
> </table>
> 
> ## Features
> ...
> ```
> 
> Example A will not work but B will work. The blank line is required by proper Markdown but the lack of it is forgiven by GitHub Markdown preview. So Example A will work in the GitHub Markdown preview (giving the wrong impression that all is well) but will not work when it is rendered into an HTML page via GitHub pages.

Yup I made sure to include a newline in between but the error still persists

**Comment by Incogdino** [comment]
> Is it directly using the `docs/README.md` file from your repository? If it isn't, you may have typo-ed something. From the screenshot you shared, it seems like the table is entirely missing and there is just a closing `</table>` tag. That might be causing parsing issues and may be the reason your markdown is not being rendered.
> <img alt="Screenshot 2024-09-17 at 08 09 10" width="806" src="https://private-user-images.githubusercontent.com/66869201/367961631-a7b245e8-71cc-4344-bdc2-b9b82c9c1d65.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY1NTA5OTYsIm5iZiI6MTcyNjU1MDY5NiwicGF0aCI6Ii82Njg2OTIwMS8zNjc5NjE2MzEtYTdiMjQ1ZTgtNzFjYy00MzQ0LWJkYzItYjliODJjOWMxZDY1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTE3VDA1MjQ1NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM0ODE4ZWM1MTNkMjIwMGEwMDdhYzhkYzM4YWJiYTU2Y2NiODA5YmQ2NDRiZTVkZWViNjRhMTBmZWMzNDk5YjImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.E3LzwmZ1qmjeSoHhI-YY2bnVaZChyduZCph1yfP5UFk">
> If that still doesn't work, try to use pipe characters for your table instead of using the `<table>` tags. [Here is how this works](https://diplodoc.com/docs/en/syntax/tables/gfm).
> 
> Basically just try to look for some inconsistency - unpaired brackets/backticks/tags, extra space etc.




On hindsight, I should've included the whole screenshot of the table under the pile of text. But yea I did make sure to close the table with the appropriate tag. I also checked and made sure that there was only one closing tag so I'm not too sure why another closing tag occurred. 

Anyways your suggestion of using pipe characters helped solve the issue. It also made the code look cleaner and less cluttered as compared to using HTML. Thanks for that! 

Do you happen to know why some GFMD elements still cannot be processed like using task lists or alerts? Are there solutions out there for this too? I can't seem to find any.

**Comment by Dinoman44** [comment]
Glad I could help!

Unfortunately I'm not too sure why tasklists and alerts are not being processed since both kramdown and GFM support them (and with the same syntax, as far as I can tell). I'll keep looking, but if anyone else has figured out the solution feel free to jump in.

**Comment by elboyran** [comment]
Hello @Dinoman44,

I have a similar issue: this [README page](https://github.com/dianna-ai/dianna/tree/main#readme) renders only after the first drop-down element (before the **Getting Started** section) on [github-pages](https://dianna-ai.github.io/dianna/):

![image](https://github.com/user-attachments/assets/ae515ef9-4a32-4b6d-91e7-04d09abbd368)

BTW there are empty lines between the sections.

Would you be able to help me as well?
